src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Preferences;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class PreferencesConfiguration implements DataConfigurationInterface
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    public function __construct(Configuration $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration()
+    {
+        return [
+            'enable_ssl' => (bool) $this->configuration->get('PS_SSL_ENABLED'),
```

Comment:
```diff
- (bool) $this->configuration->get('PS_SSL_ENABLED')
+ $this->configuration->getBoolean('PS_SSL_ENABLED')
```

See https://github.com/PrestaShop/PrestaShop/pull/8705

Reply:
@mickaelandrieu I've read the conversation on the PR, if the value is not bool here it will display incorrect style of the switch, or did I misunderstand something?

Reply:
I get you missed the point in the second line. @mickaelandrieu is suggesting to use `getBoolean` instead of a `(bool)` cast.
The discussion on the other PR was about these forgotten calls. :)

Reply:
oh right I missed that, thanks @Quetzacoalt91 :)